Tests/TuistKitTests/Models/ProjectTests.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import Basic
+@testable import TuistKit
+import XCTest
+
+final class ProjectTests: XCTestCase {
+    func test_sortedTargetsForProjectScheme() {
```

Comment:
very minor: 

similarly here leveraging a few helpers methods could help reduce the elements within the test to the essential elements being tested:

```swift
   func createTargetNodes(project: Project,
                           dependencies: [(target: Target, dependencies: [Target])]) -> [TargetNode] {
        let nodesCache = Dictionary(uniqueKeysWithValues: dependencies.map {
            ($0.target.name, TargetNode(project: project,
                                        target: $0.target,
                                        dependencies: []))
        })
        
        return dependencies.map {
            let node = nodesCache[$0.target.name]!
            node.dependencies = $0.dependencies.map { nodesCache[$0.name]! }
            return node
        }
    }
    
    func createGraph(project: Project,
                     dependencies: [(target: Target, dependencies: [Target])]) -> Graph {
        let targetNodes = createTargetNodes(project: project, dependencies: dependencies)
        
        let cache = GraphLoaderCache()
        let graph = Graph.test(cache: cache)
        
        targetNodes.forEach { cache.add(targetNode: $0) }
        
        return graph
    }
```

I'm still personally wondering if this helper is any clearer or just more confusing...

The end result would be:

```swift
   func test_sortedTargetsForProjectScheme() {
        let framework = Target.test(name: "Framework", product: .framework)
        let app = Target.test(name: "App", product: .app)
        let appTests = Target.test(name: "AppTets", product: .unitTests)
        let frameworkTests = Target.test(name: "FrameworkTests", product: .unitTests)
        let project = Project.test(targets: [
            framework, app, appTests, frameworkTests
        ])
        
        let graph = createGraph(project: project, dependencies: [
            (target: framework, dependencies: []),
            (target: frameworkTests, dependencies: [framework]),
            (target: app, dependencies: [framework]),
            (target: appTests, dependencies: [app]),
        ])

        let got = project.sortedTargetsForProjectScheme(graph: graph)
        XCTAssertEqual(got.count, 4)
        XCTAssertEqual(got[0], framework)
        XCTAssertEqual(got[1], app)
        XCTAssertEqual(got[2], appTests)
        XCTAssertEqual(got[3], frameworkTests)
    }
```

Reply:
> I'm still personally wondering if this helper is any clearer or just more confusing...

I often think about this, because really tests should be super declarative. 

But in this case you've just removed some duplication whilst still keeping it declarative - I think it's OK.

Reply:
Same here. I think having a lot of helpers introduces a lot of indirection making the test less linear and harder to debug. As long as we keep an eye and don't get crazy with helpers it should be fine. Your suggestions seem reasonable to me.